packages/beacon-node/src/chain/chain.ts

```diff
@@ -335,6 +359,7 @@ export class BeaconChain implements IBeaconChain {
 
   /** Populate in-memory caches with persisted data. Call at least once on startup */
   async loadFromDisk(): Promise<void> {
+    await this.regen.init();
```

Comment:
Dumb question: Why don't we need to `regen.init()` in the `BeaconChain` constructor?

Reply:
it's a conventional place to do async init, see the `this.opPool.fromPersisted(this.db)` call below